### ISSUE TITLE
Compilation error with CodeGenerator and custom profile

### ISSUE THREAD

**Issue by sttawm** [slick/slick#2189]
**DESCRIPTION:**

I am getting unexpected compilation errors when running the code generator with a custom profile that includes implicit `TypedType[T]`s.

I am able to fix it if, in the generated code, I manually replace the line `val profile: slick.jdbc.JdbcProfile` with `val profile: foo.CustomProfile`. But the code generator doesn't give me access to that line.

Here's a slimmed down version of the generated code: a custom profile with a custom typed type.

```scala
object Tables extends {
  val profile = foo.CustomProfile
} with Tables

trait Tables {
  val profile: slick.jdbc.JdbcProfile // <- replace this with `foo.CustomProfile` to fix compilation errors
  import profile.api._ // want this to import foo.CustomProfile._ to get implicits

  ...

  case class MyRow(foo: Foo)
  def * = (foo) <> (MyRow.tupled, MyRow.unapply) // doesn't compile
}
```

Given that I have my custom profile, something like:
```scala
trait CustomProfile extends slick.jdbc.PostgresProfile {

  object CustomApi extends API {
    implicit val customMapper: TypedType[Foo] = ...
  }
}
```

**STEPS TO REPRODUCE:**

Give the code generator a custom profile that uses a custom, implicit mapper.

**EXPECTED RESULT:**
I'd expect the implicits from `foo.CustomProfile` to be imported and the code to compile. 

**ACTUAL RESULT:**

The implicits from `foo.CustomProfile` aren't imported and the generated code doesn't compile.

**SETUP:**
Specifically, I'm using `pg-slick`s custom traits for first-class support of play-json and joda-time.

**Comment by hvesalai** [comment]
I don't believe this is an issue in Slick, but rather it is just how Scala works. You have to refactor your types.

`import profile.api._ ` can't import anything from profile that `slick.jdbc.JdbcProfile` doesn't have since that's the type.

**Comment by hvesalai** [comment]
Please do not respond to this issue unless you really think there is a bug in Slick. If you just want help from other users, then join the gitter.im channel for slack

**Comment by sttawm** [comment]
I actually still think this is a bug. Should the `CodeGenerator` really be setting the type to `slick.jdbc.JdbcProfile` when given the type `foo.CustomProfile` as its argument?

**Comment by hvesalai** [comment]
Ok, I'll re-open. Do you want to try to provide a fix for it?
